packages/cdk/click-outside/src/useClickOutside.ts

```diff
@@ -0,0 +1,53 @@
+import type { ComponentPublicInstance, Directive } from 'vue'
+
+import { isHTMLElement, off, on } from '@idux/cdk/utils'
+
+type ClickOutside = () => void
+
+type UseClickOutside = () => Directive<HTMLElement | ComponentPublicInstance, ClickOutside>
+
+type FnMap = WeakMap<HTMLElement, ClickOutside>
+type SelfMap = WeakMap<HTMLElement, (e: MouseEvent) => void>
+
+const useClickOutside: UseClickOutside = () => {
+  const fnMap: FnMap = new WeakMap()
+  const selfMap: SelfMap = new WeakMap()
+
+  return {
+    mounted(el, binding) {
+      const element = isHTMLElement(el) ? el : el.$el
+      const fn = () => {
+        binding.value?.()
+      }
+      on(window, 'click', fn)
+      const stopPropagation = (e: MouseEvent) => e.stopPropagation()
+      on(element, 'click', stopPropagation)
+      fnMap.set(element, fn)
+      selfMap.set(element, stopPropagation)
+    },
+    updated(el, binding) {
+      const element = isHTMLElement(el) ? el : el.$el
+      const fn = () => {
+        binding.value?.()
+      }
+      const oldFn = fnMap.get(element)!
+      if (fn === oldFn) {
+        return
+      }
+      off(window, 'click', oldFn)
+      on(window, 'click', fn)
+      fnMap.set(element, fn)
+    },
+    beforeUnmount(el) {
```

Comment:
```suggestion
    unmounted(el) {
```

所有的资源释放操作统一放到 unmounted 中